dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/route/RouteService.java

```diff
@@ -142,9 +144,19 @@ public ResponseEntity<String> exec( Route route, User user, HttpServletRequest r
         HttpHeaders queryParameters = new HttpHeaders();
         request.getParameterMap().forEach( ( key, value ) -> queryParameters.addAll( key, List.of( value ) ) );
 
-        UriComponentsBuilder uriComponentsBuilder = UriComponentsBuilder.fromHttpUrl( route.getUrl() )
+        UriComponentsBuilder uriComponentsBuilder = UriComponentsBuilder.fromHttpUrl( route.getBaseUrl() )
             .queryParams( queryParameters );
 
+        if ( subPath.isPresent() )
+        {
+            if ( !route.allowsSubpaths() )
+            {
+                throw new IllegalArgumentException(
```

Comment:
We try to use the exceptions that already map to a HTTP response directly, in this case use `org.hisp.dhis.feedback.BadRequestException`. Then propagate the `throws` on the methods to the controller level.

Reply:
@jbee thanks for the review! - most services seem to return IllegalArgumentException or similar which makes sense to me, as the service might be used differently in different controllers, can you expand on why we shouldn't keep concerns separated between service and controller here?  Happy to change it if needed, but curious why we're moving in that diretion